compiler/passes/convert-uast.cpp

```diff
@@ -3837,12 +3907,27 @@ struct Converter {
       initExpr = convertExprOrNull(node->initExpression());
     }
 
-    auto ret = new DefExpr(varSym, initExpr, typeExpr);
+    auto def = new DefExpr(varSym, initExpr, typeExpr);
+    VariableDefInfo ret = { def, def };
+
+    auto loopFlags = LoopAttributeInfo::fromVariableDeclaration(context, node);
+    if (!loopFlags.empty()) {
+      auto block = new BlockStmt(BLOCK_SCOPELESS);
+      block->insertAtTail(new CallExpr(PRIM_GPU_ATTRIBUTE_BLOCK));
+      block->insertAtTail(def);
+
+      auto primBlock = new BlockStmt(BLOCK_SCOPELESS);
+      block->insertAtTail(primBlock);
+      primBlock->insertAtTail(new CallExpr(PRIM_GPU_PRIMITIVE_BLOCK));
+      loopFlags.insertPrimitives(*this, primBlock);
+
+      ret.entireExpr = block;
+    }
```

Comment:
Consider adding an `else` block that does `ret.entireExpr = def` to make the code easier to follow. (and changing your initial definition of `ret` above)